frontend/src/lib/components/PropertyFilters/components/UnifiedPropertyFilter.tsx

```diff
@@ -124,13 +129,21 @@ export function UnifiedPropertyFilter({ index, onComplete, logic }: PropertyFilt
                     </>
                 )
             },
-            dataSource:
-                eventProperties.map(({ value: eventValue, label, is_numerical }) => ({
-                    name: label || '',
-                    key: `event_${eventValue}`,
-                    eventValue,
-                    is_numerical,
-                })) || [],
+            dataSource: [],
+            async: featureFlags['property-definitions-search'],
+            totalCount: eventPropertyCount,
+            loadMoreItems: async () => {
+                const initial = eventProperties.length === 0
+                await loadPropertyDefinitions(initial)
```

Comment:
Actions return immediately and their listeners run in the background. Awaiting an action does nothing.